### PR TITLE
Adds error handling for group name length

### PR DESCRIPTION
**Pull request by ljewalsh**
Also refactored how we were handling errors in general

### CODE REVIEW

File: src/requests/addGroup.spec.js [hunk 1]
```diff
@@ -32,3 +33,11 @@ test(`create a new tracked link`, async (t) => {
 
   t.is(newGroupId, NEW_GROUP_ID)
 })
+
+test(`GroupNameCharLimitExceeded is thrown when the group name is too long`, async (t) => {
+  const GROUP_NAME = 'hereisareallyreallylonggroupname'
+
+  const error = await t.throwsAsync(addGroup(GROUP_NAME))
```
Comment (stayradiated) [inline src/requests/addGroup.spec.js]:
```suggestion
  const error = await t.throwsAsync(
    addGroup(GROUP_NAME),
    GroupNameCharLimitExceeded
  )
```

Comment (stayradiated) [reply src/requests/addGroup.spec.js]: You can pass an error class as the second argument to `throwsAsync`

File: src/requests/addLinkToGroup.spec.js [hunk 2]
```diff
@@ -37,5 +37,6 @@ test('If genius replies with 200 but LinkResponses contains an error message, th
       }]
     })
 
-  await t.throws(addLinkToGroup(LINK, GROUP_ID), 'Genius API could not generate a code, but still came back with 200')
+  const error = await t.throwsAsync(addLinkToGroup(LINK, GROUP_ID))
+  t.truthy(error instanceof AddLinkToGroupFailed)
```
Comment (stayradiated) [inline src/requests/addLinkToGroup.spec.js]: I would suggest you use `t.true` here -- as `instanceof` returns a boolean.

But in this case, I would also suggest passing `AddLinkToGroupFailed` directly to `throwsAsync` as the second argument.

File: src/requests/addGroup.spec.js (hunk 1 above)
Comment (ljewalsh) [reply src/requests/addGroup.spec.js]: Sure sure

File: src/requests/addLinkToGroup.spec.js (hunk 2 above)
Comment (ljewalsh) [reply src/requests/addLinkToGroup.spec.js]: ok thanks!

**Pull request merged**